tests/integration/test.replication.js

```diff
@@ -3609,6 +3609,37 @@ adapters.forEach(function (adapters) {
       });
     });
 
+    it('#3658 replication with ddoc filter, update doc', function() {
+      var db = new PouchDB(dbs.name);
+      var remote = new PouchDB(dbs.remote);
+
+	      return remote.bulkDocs([{
```

Comment:
It seems the bug is that you are putting the design doc in the remote database. If you are replicating *from* the local database, then that is where the ddoc should go. :)